devices/Hdc1080/Hdc1080.cs

```diff
@@ -0,0 +1,173 @@
+        /// </summary>
+        /// <returns>
+        /// Manufacturer id
+        /// </returns>
+        public string ReadManufacturerId() => ReadData(RegisterAddress.ManufacturerId).ToString();
+
+        /// <summary>
+        /// Reads serial number, check data sheet, page 14, 8.6.4 section
+        /// </summary>
+        /// <returns>
+        /// Serial number
+        /// </returns>
+        public string ReadSerialNumber() => $"{ReadData(RegisterAddress.SerialIdFirstByte)}{ReadData(RegisterAddress.SerialIdSecondByte)}{ReadData(RegisterAddress.SerialIdThirdByte)}";
+
+        /// <summary>
+        /// Sets humidity resolution
+        /// </summary>
+        /// <param name="resolution">Humidity resolution</param>
+        public void SetHumidityResolution(HumidityResolution resolution)
+        {
+            ConfigurationRegister register = (ConfigurationRegister)ReadRegister();
+            register.HumidityMeasurementResolution = (byte)resolution;
+
+            WriteRegister(register);
+        }
+
+        /// <summary>
+        /// Sets temperature resolution
+        /// </summary>
+        /// <param name="resolution">Temperature resolution</param>
+        public void SetTemperatureResolution(TemperatureResolution resolution)
+        {
+            ConfigurationRegister register = (ConfigurationRegister)ReadRegister();
+            register.TemperatureMeasurementResolution = (byte)resolution;
+
+            WriteRegister(register);
+        }
+
+        /// <summary>
+        /// Runs heater for given number of milliseconds, check data sheet, page 9, 8.3.3 section
+        /// </summary>
+        /// <param name="milliseconds">Number of milliseconds</param>
+        public void HeatUp(int milliseconds)
+        {
+            SetHeater(true);
+            for (int i = 1; i < (milliseconds); i++)
+            {
+                _i2cDevice.Write(new byte[] { (byte)RegisterAddress.TemperatureMeasurement });
+                // wait SCL free
+                Thread.Sleep(20);
+                _i2cDevice.Read(new byte[4]);
+            }
+            SetHeater(false);
+        }
+
+        /// <summary>
+        /// Cleanup
+        /// </summary>
+        public void Dispose()
+        {
+            _i2cDevice?.Dispose();
+            _i2cDevice = null!;
+        }
+        private long ReadData(RegisterAddress address)
+        {
+            byte[] resultData = new byte[2];
+            _i2cDevice.Write(new byte[] {(byte) address });
+           //  Thread.Sleep(9);
+            _i2cDevice.Read(resultData);
+            return resultData[0] << 8 | resultData[1];
+        }
+
+        private void WriteRegister(ConfigurationRegister register)
+        {
+            _i2cDevice.Write(new byte[] { (byte)RegisterAddress.Configuration, register.GetData(), (byte)RegisterAddress.TemperatureMeasurement });
+            // wait SCL free
+            Thread.Sleep(20);
```

Comment:
Is 20 from the documentation?

Reply:
Doc says at most 15ms but who knows how much you can trust this so I put 20